src/Identity/UI/src/Microsoft.AspNetCore.Identity.UI.csproj

```diff
@@ -47,113 +43,22 @@
-    <UIFrameworkVersionMoniker Include="V4" />
-  </ItemGroup>
-
-  <ItemGroup>
-    <Folder Include="build\" />
-  </ItemGroup>
-
-  <!-- Source build doesn't build this package -->
-  <Target Name="BuildRazorViews" DependsOnTargets="Compile" BeforeTargets="Build" Condition="'$(DotNetBuildFromSource)' != 'true'">
-    <Message Text="Building razor views assemblies" Importance="High" />
-    <MSbuild Projects="$(MSBuildThisFile)"
-             Targets="BuildForUI"
-             Properties="
-             BuildProjectReferences=false;
-             RazorCompileOnBuild=true;
-             UIFrameworkVersion=%(UIFrameworkVersionMoniker.Identity);
-             IntermediateOutputPath=$(IntermediateOutputPath)%(UIFrameworkVersionMoniker.Identity)\;" />
-
-    <Copy
-      SourceFiles="$(IntermediateOutputPath)%(UIFrameworkVersionMoniker.Identity)\$(TargetName).Views.%(UIFrameworkVersionMoniker.Identity).dll"
-      DestinationFolder="$(OutDir)"
-      SkipUnchangedFiles="$(SkipCopyUnchangedFiles)"
-      OverwriteReadOnlyFiles="$(OverwriteReadOnlyFiles)"
-      Retries="$(CopyRetryCount)"
-      RetryDelayMilliseconds="$(CopyRetryDelayMilliseconds)"
-      UseHardlinksIfPossible="$(CreateHardLinksForCopyFilesToOutputDirectoryIfPossible)"
-      UseSymboliclinksIfPossible="$(CreateSymbolicLinksForCopyFilesToOutputDirectoryIfPossible)">
-
-      <Output TaskParameter="DestinationFiles" ItemName="FileWrites"/>
-      <Output TaskParameter="DestinationFiles" ItemName="_RazorAssembly"/>
-
-    </Copy>
-
-    <Copy
-      SourceFiles="$(IntermediateOutputPath)%(UIFrameworkVersionMoniker.Identity)\$(TargetName).Views.%(UIFrameworkVersionMoniker.Identity).pdb"
-      DestinationFolder="$(OutDir)"
-      SkipUnchangedFiles="$(SkipCopyUnchangedFiles)"
-      OverwriteReadOnlyFiles="$(OverwriteReadOnlyFiles)"
-      Retries="$(CopyRetryCount)"
-      RetryDelayMilliseconds="$(CopyRetryDelayMilliseconds)"
-      UseHardlinksIfPossible="$(CreateHardLinksForCopyFilesToOutputDirectoryIfPossible)"
-      UseSymboliclinksIfPossible="$(CreateSymbolicLinksForCopyFilesToOutputDirectoryIfPossible)">
-
-      <Output TaskParameter="DestinationFiles" ItemName="FileWrites" />
-    </Copy>
-
-    <Message Importance="High" Text="$(MSBuildProjectName) -&gt; %(_RazorAssembly.FullPath)" />
-
-  </Target>
-
-  <PropertyGroup>
-    <RazorTargetName>Microsoft.AspNetCore.Identity.UI.Views.$(UIFrameworkVersion)</RazorTargetName>
-  </PropertyGroup>
-
-  <Target Name="SetupRazorInputs">
+  <Target Name="SetupRazorInputs" BeforeTargets="ResolveRazorGenerateInputs">
     <ItemGroup>
-      <_RazorGenerate Include="Areas\Identity\Pages\$(UIFrameworkVersion)\**\*.cshtml" />
+      <_RazorGenerate Include="Areas\Identity\Pages\V4\**\*.cshtml" />
 
       <RazorGenerate Include="@(_RazorGenerate)" Link="Areas\Identity\Pages\%(RecursiveDir)%(Filename)%(Extension)" />
     </ItemGroup>
   </Target>
 
-  <Target Name="BuildForUI" DependsOnTargets="SetupRazorInputs;RazorCompile" />
```

Comment:
FYI @javiercn. The `RazorCompile` target no longer runs for 6.0 (compilation of Razor files is part of Csc). I didn't spend a lot of time trying to continue supporting multitargeting here.

Reply:
I'm somewhat ok with this, but we are going to have to figure an alternative for the future. We are likely upgrading to Bootstrap5 and I'm not sure if we'll have to keep 4 around to ease migration.